ggshield/cmd/honeytoken/create_with_context.py

```diff
@@ -0,0 +1,147 @@
+from collections import Counter
+from pathlib import Path
+from typing import Any, Optional
+
+import click
+from pygitguardian.models import Detail, HoneytokenWithContextResponse
+
+from ggshield.cmd.honeytoken.utils import generate_random_honeytoken_name
+from ggshield.cmd.utils.common_options import add_common_options
+from ggshield.cmd.utils.context_obj import ContextObj
+from ggshield.core.client import create_client_from_config
+from ggshield.core.errors import UnexpectedError
+from ggshield.utils.click import RealPath
+
+
+def _list_extensions_in_dir(
+    dir: Path, max_depth: int = 3, depth: int = 0
+) -> "list[str]":
```

Comment:
nit (not a blocker): you should be able to remove the quotes around `list[str]` here.